modules/stanford_courses_tag_translate/stanford_courses_tag_translate.pages.inc

```diff
@@ -0,0 +1,531 @@
+
+  // If not a valid param then we should stop doing more.
+  if (!is_numeric($ctid)) {
+    drupal_set_message("Invalid course tag translation id.", "error");
+    drupal_not_found();
+    exit();
+  }
+
+  // Get the translation from the DB. Just to be sure.
+  $tobj = stanford_courses_tag_translate_get_translation($ctid);
+
+  // If it wasn't there then die.
+  if (!$tobj) {
+    drupal_set_message("Invalid course tag translation id.", "error");
+    drupal_not_found();
+    exit();
+  }
+
+  // Put the value in the form so we can use it to delete stuff later.
+  $form["ctid"] = array(
+    "#type" => "value",
+    "#value" => $ctid,
+  );
+
+  // Prompt the user to confirm their choices in life.
+  $question = t("Are you sure you want to remove this tag translation");
+  $path = "admin/config/stanford/courses/tag-translate";
+  $description = t("This action cannot be un-done. Choose wisely.");
+  $yes = t("Yes, delete this forever");
+
+  $form = confirm_form($form, $question, $path, $description, $yes);
+  return $form;
+}
+
+/**
+ * Confirm delete submit hook.
+ *
+ * If the user confirms they want to delete the tag translation we should
+ * probably actually do that thing we asked them about.
+ *
+ * @param array $form
+ *   The form array.
+ * @param array $form_state
+ *   The form state array.
+ *
+ */
+function stanford_courses_tag_translate_delete_form_submit($form, &$form_state) {
+
+  // Get the ctid from the form submission.
+  $values = $form_state["values"];
+  $ctid = check_plain($values["ctid"]);
+
+  // With fire!
+  db_delete("stanford_courses_tag")
+    ->condition("ctid", $ctid)
+    ->execute();
+
+  // Ok, done. Now go back to the main page.
+  drupal_set_message("Tag translation was successfully removed.", "status");
+  drupal_goto("admin/config/stanford/courses/tag-translate");
+}
+
+/**
+ * Import tag translations form.
+ *
+ * Get and render the import form. Also, provide a helpful little message saying
+ * they are going to blow away everything that already exists.
+ *
+ */
+function stanford_courses_tag_translate_import() {
+  $render = array();
+
+  // Caution: Like, for realz.
+  drupal_set_message("WARNING: Importing new values will empty and replace all existing values. This action is not recoverable.", "warning");
+
+  // Get basic form block.
+  $render["theform"] = drupal_get_form("stanford_courses_tag_translate_export_form");
+  $render["theform"]["txtxprt"]['#title'] = t("Import");
+
+  // Done done delta 58.
+  return $render;
+}
+
+/**
+ * Validate the import submission.
+ *
+ * Don't trust your users and don't trust their input. Check to see if what
+ * they are sending us is in a valid json format.
+ *
+ * @param array $form
+ *   The form array.
+ * @param array $form_state
+ *   The form state array.
+ */
+function stanford_courses_tag_translate_import_form_validate($form, &$form_state) {
+
+  // Get the submission value.
+  $values = $form_state["values"];
+  $json = filter_xss($values["txtxprt"]);
```

Comment:
Of the many string sanitization functions available, why do you choose `filter_xss()` here? Asking for my own edification, not questioning your decision.